wire-ios-sync-engine/Source/Use cases/SnoozeCertificateEnrollmentUseCase.swift

```diff
@@ -72,10 +74,14 @@ final class SnoozeCertificateEnrollmentUseCase: SnoozeCertificateEnrollmentUseCa
             id: actionId,
             interval: interval
         ) {
-            // We save the end of the grace period once and should not update it.
-            let notificationObject = FeatureRepository.FeatureChange.e2eIEnabled(gracePeriod: nil)
-            NotificationCenter.default.post(name: .featureDidChangeNotification,
-                                            object: notificationObject)
+            if isUpdateMode {
+                NotificationCenter.default.post(name: .checkForE2EICertificateExpiryStatus, object: nil)
+            } else {
+                // We save the end of the grace period once and should not update it.
+                let notificationObject = FeatureRepository.FeatureChange.e2eIEnabled(gracePeriod: nil)
+                NotificationCenter.default.post(name: .featureDidChangeNotification,
+                                                object: notificationObject)
+            }
         }
 
         recurringActionService.registerAction(recurringAction)
```

Comment:
All the other actions are registered on the main thread. I wonder if we should add something like ...
```suggestion
        await MainActor.run {
            recurringActionService.registerAction(recurringAction)
        }
```

Reply:
done